scipy/integrate/_ivp/tests/test_ivp.py

```diff
@@ -1091,3 +1091,79 @@ def test_initial_state_finiteness(f0_fill):
     msg = "All components of the initial state `y0` must be finite."
     with pytest.raises(ValueError, match=msg):
         solve_ivp(fun_zero, [0, 10], np.full(3, f0_fill))
+
+@pytest.mark.parametrize('method', ['RK23', 'RK45', 'DOP853', 'Radau', 'BDF'])
+def test_tbound_respected_small_interval(method):
+    """Regression test for gh-17341"""
+    SMALL = 1e-4
+
+    # f[y(t)] = 2y(t) on t in [0,SMALL]
+    #           undefined otherwise 
+    def f(t,y):
+        if t>SMALL:raise ValueError("Function was evaluated outside interval")
```

Comment:
Don't want to be overly picky, but the code looks a bit untidy.

Please use standard `if` formatting, add spaces around all operators, use `0.0` to denote floats.

Reply:
Hopefully now better; most of this code came from the various bug reports.